src/Data/LLVM/BitCode/BitString.hs

```diff
@@ -33,35 +40,62 @@ pattern Bytes' n = NumBytes n
 bitCount :: NumBits -> Int
 bitCount (NumBits n) = n
 
+{-# INLINE addBitCounts #-}
 addBitCounts :: NumBits -> NumBits -> NumBits
 addBitCounts (NumBits a) (NumBits b) = NumBits $ a + b
 
+{-# INLINE subtractBitCounts #-}
 subtractBitCounts :: NumBits -> NumBits -> NumBits
 subtractBitCounts (NumBits a) (NumBits b) = NumBits $ a - b
 
+{-# INLINE bytesToBits #-}
+bitsToBytes :: NumBits -> (NumBytes, NumBits)
+bitsToBytes (NumBits n) = (NumBytes $ n `shiftR` 3, NumBits $ n .&. 7)
+
+{-# INLINE bitsToBytes #-}
+bytesToBits :: NumBytes -> NumBits
+bytesToBits (NumBytes n) = NumBits $ n `shiftL` 3
+
+----------------------------------------------------------------------
+
 data BitString = BitString
   { bsLength :: !NumBits
   , bsData   :: !Integer
-  } deriving Show
+  } deriving (Show, Eq)
+
+-- | Create an empty BitString
 
-instance Eq BitString where
-  BitString n i == BitString m j = n == m && i == j
+emptyBitString :: BitString
+emptyBitString = BitString (NumBits 0) 0
 
-instance Semigroup BitString where
-  BitString n@(NumBits n') i <> BitString m j =
-    BitString (addBitCounts n m) (i .|. (j `shiftL` n'))
 
-instance Monoid BitString where
-  mempty = BitString (NumBits 0) 0
-  mappend = (<>)
+-- | Join two BitString representations together to form a single larger
+-- BitString.  The first BitString is the "lower" value portion of the resulting
+-- BitString.
+
```

Comment:
A minor thing, but why put whitespace between a function and its Haddocks? It's somewhat odd to see them separated like this, especially when the opposite convention is so popular.

Reply:
It's a readability habit I have since I find it helps me see the function declaration and corresponding types more easily because it isn't tucked right up against the docs.  That's just the background: I'm not strongly committed to this though, so I'll try to shift back towards the convention going forward.